dallinger/config.py

```diff
@@ -0,0 +1,163 @@
+from __future__ import unicode_literals
+
+from collections import deque
+from ConfigParser import SafeConfigParser
+import distutils.util
+import os
+import threading
+
+marker = object()
+
+try:
+    unicode = unicode
+except NameError:  # Python 3
+    unicode = str
+
+
+class Configuration(object):
+
+    SUPPORTED_TYPES = {
+        bytes,
+        unicode,
+        int,
+        float,
+        bool,
+    }
+
+    def __init__(self):
+        self.data = deque()
+        self.types = {}
+        self.synonyms = {}
+        self.ready = False
+
+    def extend(self, mapping, cast_types=False):
+        normalised_mapping = {}
+        for key, value in mapping.items():
+            key = self.synonyms.get(key, key)
+            if key not in self.types:
+                # This key hasn't been registered, we ignore it
+                continue
```

Comment:
I think it would be better to raise an exception when trying to set a key of unknown type, rather than silently ignoring it. Ignoring it means users won't be notified when they make a typo.